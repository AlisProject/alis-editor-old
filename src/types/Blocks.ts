export enum BlockType {
  Rule = 'Rule',
  Text = 'Text',
  Paragraph = 'Paragraph',
  Image = 'Image',
  Quote = 'Quote',
  Heading = 'Heading',
  Embed = 'Embed'
}

export interface Block {
  id: string
  type: BlockType
  payload?: any
  children?: Block[]
}

export interface RuleBlock extends Block {
  type: BlockType.Rule
}

export interface EmbedBlock extends Block {
  type: BlockType.Embed,
  payload: {
    src: string
  }
}

export interface ParagraphBlock extends Block {
  type: BlockType.Paragraph
  payload: {
    body: string
  }
}

export interface TextBlock extends Block {
  type: BlockType.Text
  payload: {
    body: string
  }
}

export interface ImageBlock extends Block {
  type: BlockType.Image
  payload: {
    src: string
    align?: string
  }
  children: Block[]
}

export interface HeadingBlock extends Block {
  type: BlockType.Heading
  payload: {
    size: 'h2' | 'h3'
    body: string
  }
  children: Block[]
}

export interface QuoteBlock extends Block {
  type: BlockType.Quote
  children: Block[]
}

/*
「詳細は [こちら](https://example.com) から確認できます」を再現する場合
[
  {
    type: 'Paragraph',
    children: [
      {
        type: 'Text',
        payload: {
          body: '詳細は'
        }
      },
      {
        type: 'Link',
        payload: {
          href: 'https://example.com'
        },
        children: [
          {
            type: 'Text',
            payload: {
              body: 'こちら'
            }
          }
        ]
      },
      {
        type: 'Text',
        payload: {
          body: 'から確認できます'
        }
      }
    ]
  }
]

*/
