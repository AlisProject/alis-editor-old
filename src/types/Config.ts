export interface AlisEditorConfig {
  articleId: string
  preview: boolean
  iframelyApikey: string
  uploadEndpoint: string
}
