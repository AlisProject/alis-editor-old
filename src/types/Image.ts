export interface ImageUploadRequest {
  article_image: string
}

export interface ImageUploadResponse {
  image_url: string
}
