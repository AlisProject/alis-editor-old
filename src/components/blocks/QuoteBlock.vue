<template>
  <blockquote>
    <component
      :is="`${childBlock.type}Block`"
      :block="childBlock"
      v-for="(childBlock, i) in block.children"
      :key="childBlock.id"
      @input="handleInput(i, $event)"
      @delete="handleDelete(i)"
      :preview="preview"
    />
  </blockquote>
</template>

<script lang="ts">
import Vue from 'vue'
import uuid from 'uuid/v4'
import { Block, QuoteBlock } from '../../types/Blocks'
import TextBlock from './TextBlock.vue'
import { cloneDeep } from 'lodash'

export default Vue.extend({
  components: {
    TextBlock
  },
  props: {
    block: Object as () => QuoteBlock,
    preview: {
      default: false,
      type: Boolean
    }
  },
  methods: {
    handleInput(idx: number, value: string) {
      const { block } = this
      block.children[idx].payload.body = value
      this.$emit('update', cloneDeep(block))
    },
    handleDelete(idx: number) {
      this.$emit('delete', this.block)
    }
  }
})
</script>

<style scoped>
blockquote {
  font-family: 'YuGothic', 'Yu Gothic';
  background: #f6f6f6;
  font-size: 16px;
  letter-spacing: 0.1em;
  line-height: 1.5;
  margin: 0.5em 0;
  padding: 20px 20px 15px;
  word-break: break-word;
}

blockquote textarea {
  font-family: 'YuGothic', 'Yu Gothic';
  font-size: 16px;
  letter-spacing: 0.1em;
  line-height: 1.75;
  margin: 0;
  word-break: break-word;
}

@media (max-width: 640px) {
  blockquote {
    font-size: 14px;
  }
}
</style>
