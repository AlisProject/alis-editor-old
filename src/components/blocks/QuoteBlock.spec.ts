import { mount, config } from '@vue/test-utils'
import QuoteBlock from './QuoteBlock.vue'
import * as BlockTypes from '../../types/Blocks'
config.logModifiedComponents = false

describe('QuoteBlock.vue', () => {
  test('renders correctly', () => {
    const block: BlockTypes.QuoteBlock = {
      id: '123',
      type: BlockTypes.BlockType.Quote,
      children: []
    }
    const wrapper = mount(QuoteBlock, {
      propsData: {
        block
      }
    })
    expect(wrapper.element).toMatchSnapshot()
  })
})
