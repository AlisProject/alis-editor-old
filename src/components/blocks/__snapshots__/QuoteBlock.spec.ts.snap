// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuoteBlock.vue renders correctly 1`] = `<blockquote />`;
